[3437] Migrate to MUI 5

AresEkb opened the pull request:
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

## Review

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

AresEkb committed:
[3437] Migrate to MUI 5

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3437
Signed-off-by: Denis Nikiforov <denis.nikif@gmail.com>
Signed-off-by: Michaël Charfadi <michael.charfadi@obeosoft.com>
Signed-off-by: Stéphane Bégaudeau <stephane.begaudeau@obeo.fr>

AresEkb commented:
Two Cypress-tests are unstable:
- `cypress/e2e/project/forms/widget-reference.cy.ts`
- `cypress/e2e/project/portals/portals.cy.ts`

They sometimes fail. But I can't reproduce it locally.

gcoutable commented:
There are still references to `material-ui@v4` in `package-lock.json`.
Can you generate `package-lock.json` from scratch?

AresEkb commented:
> There are still references to `material-ui@v4` in `package-lock.json`. Can you generate `package-lock.json` from scratch?

I've regenerated. But there are still Material UI 4 dependencies because of gantt-task-react

Created a pull request to migrate it to MUI 5 https://github.com/ObeoNetwork/gantt-task-react/pull/9

Probably it can be made in 3 stages:
1) Migrate Sirius Web to MUI 5
2) Migrate gantt-task-react to MUI 5
3) Migrate Sirius Web to new gantt-task-react

Or in two stages:
1) Migrate gantt-task-react to MUI 5
2) Migrate Sirius Web to MUI 5 and new gantt-task-react

gcoutable commented:
> I've regenerated. But there are still Material UI 4 dependencies because of gantt-task-react
> 
> Created a pull request to migrate it to MUI 5 [ObeoNetwork/gantt-task-react#9](https://github.com/ObeoNetwork/gantt-task-react/pull/9)
> 
> Probably it can be made in 3 stages:
> 
>     1. Migrate Sirius Web to MUI 5
> 
>     2. Migrate gantt-task-react to MUI 5
> 
>     3. Migrate Sirius Web to new gantt-task-react
> 
> 
> Or in two stages:
> 
>     1. Migrate gantt-task-react to MUI 5
> 
>     2. Migrate Sirius Web to MUI 5 and new gantt-task-react

We will do it in two stages.

Thank you for your contribution to gantt-task-react, I will look at it when I will have the time.

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json:
The list of widgets on the left of a FormDesciptionEditor representation contains icons with the wrong color.
![image](https://github.com/user-attachments/assets/65b2a6d9-14de-49fd-907a-1f22b585e2f4)

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ButtonWidget.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
The widget button has a different style (also in modals for creating representation where it's more noticeable)
After
![image](https://github.com/user-attachments/assets/fa6ef564-5be0-42f2-8a89-bfc45e574706)
Before
![image](https://github.com/user-attachments/assets/72c60ff2-8421-4c6c-8fc5-16d68d833866)

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/CheckboxWidget.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
The checkbox and radio widget are not correctly align 
![image](https://github.com/user-attachments/assets/3517ebd5-7c78-499c-8656-9fd2c35647e7)
This is especially visible on the detail view
![image](https://github.com/user-attachments/assets/7f5feff6-cf4a-4983-81de-e3a81d1790ae)

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/ImageWidget.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
The image preview does to have the same style
![image](https://github.com/user-attachments/assets/ef836c28-b918-4b3e-979f-af12c4d52e99)

Before :
![image](https://github.com/user-attachments/assets/f0026ca5-b917-41e8-b3a9-99ca6661da38)

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RadioWidget.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Same issue as the checkbox widget

mcharfadi commented on packages/forms/frontend/sirius-components-forms/package.json:
```diff
@@ -35,10 +35,10 @@
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
     "@eclipse-sirius/sirius-components-charts": "*",
+    "@emotion/react": "11.11.4",
```
Why the added dependency `"@emotion/react": "11.11.4",` here ?

mcharfadi commented on packages/forms/frontend/sirius-components-forms/src/pagelist/PageList.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
The items on the tabs are not spaced enough now
![image](https://github.com/user-attachments/assets/76a948ed-4834-4e8a-800f-f1df810b3d05)
Before :
![image](https://github.com/user-attachments/assets/365c2b9d-4ef1-4ea7-9e4f-20e6085c6f5f)

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/TreeWidget.tsx:
```diff
@@ -12,17 +12,17 @@
  *******************************************************************************/
```
The checkbox of the Tree widget lost its style and also each treeItems is now too large
After
![image](https://github.com/user-attachments/assets/995d71c1-a5ce-4416-a28b-2332693ac60f)
Before
![image](https://github.com/user-attachments/assets/0c311179-4b6d-4931-84dd-9dd85e0aba9b)

mcharfadi replied:
![image](https://github.com/user-attachments/assets/5dfd28f0-fe7e-45a5-bbaf-d7fa963846e1)
Another example

mcharfadi replied:
Seems to be the padding of MuiTreeItem-content

mcharfadi commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
We should probably increase the spacing between the buttons of the RichText widget
After
![image](https://github.com/user-attachments/assets/d962f1f1-19a2-4695-9150-e39c29870cd4)
Before
![image](https://github.com/user-attachments/assets/3ec9e159-e100-4c31-b245-6d00799613ee)

mcharfadi commented on packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx:
```diff
@@ -25,8 +25,8 @@ import {
   PropertySectionLabel,
```
The ReferenceWidget buttons on the far right are too close to each others
![image](https://github.com/user-attachments/assets/3c3265a2-e7d9-49f3-aa56-ee0128139376)
Also when a value is set the background of the value is now transparent and too small
![image](https://github.com/user-attachments/assets/7d63b721-5501-4c11-b367-503455e873a7)
Before :
![image](https://github.com/user-attachments/assets/6c005fad-065e-406f-aae7-68b006d06f41)

mcharfadi replied:
The button are now correctly placed but they have the wrong color
![image](https://github.com/user-attachments/assets/8e1aef18-9a58-4269-9394-e4ca4ddc0fba)

mcharfadi commented on packages/sirius-web/frontend/sirius-web-application/src/diagrams/DiagramFilter.tsx:
```diff
@@ -11,18 +11,18 @@
  *     Obeo - initial API and implementation
```
The filter elements modal (accessible from the last button of the diagram panel) is not positioned under the button anymore
![image](https://github.com/user-attachments/assets/89322682-8b72-4d21-8dde-2198e2171607)

mcharfadi commented on packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/list-projects-area/ListProjectsArea.tsx:
```diff
@@ -11,15 +11,15 @@
  *     Obeo - initial API and implementation
```
The name of a project is always underlined on the list. Whereas before it was underlined if it was hovered
![image](https://github.com/user-attachments/assets/60cebd1b-4f6a-406a-958b-7b652ef82e14)

mcharfadi commented on packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/workbench-views/RelatedElementsView.tsx:
```diff
@@ -12,13 +12,13 @@
  *******************************************************************************/
 import { useSelection, WorkbenchViewComponentProps } from '@eclipse-sirius/sirius-components-core';
```
The style of the buttons on the top of the Related views does not make it clear what is the state of the button
![image](https://github.com/user-attachments/assets/7be3d2a1-7cf9-410e-aa11-7aa3066df9dd)

mcharfadi commented on packages/validation/frontend/sirius-components-validation/src/ValidationView.tsx:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
![ValidationView](https://github.com/user-attachments/assets/a3f9b08a-8717-47b1-9a23-ead2f9848927)
I got a weird behavior when opening this view

mcharfadi reviewed (dismissed):
I listed some issues that would need to be addressed before merging.
I'm about to review https://github.com/ObeoNetwork/gantt-task-react/pull/9 and then I will try to fixe the above.

AresEkb commented:
During one of rebases I forgot to update dependencies from MUI 4 to 5 in TreeItemAction.tsx. That was the reason of most style problems. Conflicts between MUI 4 and 5 can cause such a problems. It should be fine now

I've also fixed menu position, project link underline and removed dependency on @emotion/react

mcharfadi commented:
Thanks you @AresEkb, this indeed fixed most issues.
I'll release a new version of [gantt-task-react](https://github.com/ObeoNetwork/gantt-task-react) now before doing what should be the last review.

sbegaudeau commented on packages/deck/frontend/sirius-components-deck/package.json:
```diff
@@ -33,8 +33,8 @@
     "@ObeoNetwork/react-trello": "2.4.11",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/icons-material": "5.15.19",
```
The new peer-dependency `tss-react` is missing here (and I suspect in other package.json files). It's also missing in the devDependencies part.

sbegaudeau replied:
Done

sbegaudeau commented on packages/diagrams/frontend/sirius-components-diagrams/package.json:
```diff
@@ -32,8 +32,8 @@
     "@eclipse-sirius/sirius-components-core": "*",
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
```
peerDependencies and devDependencies to update

sbegaudeau replied:
Done

sbegaudeau commented on packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/package.json:
```diff
@@ -36,8 +36,9 @@
-    "@material-ui/icons": "4.11.3",
+    "@mui/icons-material": "5.15.19",
+    "@mui/material": "5.15.19",
+    "@mui/x-tree-view": "7.6.1",
```
peer and dev dependencies to be updated

sbegaudeau replied:
Done

sbegaudeau commented on packages/forms/frontend/sirius-components-forms/package.json:
```diff
@@ -36,9 +36,8 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
```
peer/dev dependencies

sbegaudeau replied:
Done

sbegaudeau commented on packages/forms/frontend/sirius-components-forms/src/propertysections/SliderPropertySection.tsx:
```diff
@@ -10,9 +10,10 @@
 import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
-import Slider from '@material-ui/core/Slider';
+import Slider from '@mui/material/Slider';
+import gql from 'graphql-tag';
```
gql should be imported from Apollo client, we should probably ban import from graphql-tag

sbegaudeau replied:
Done

sbegaudeau commented on packages/forms/frontend/sirius-components-widget-reference/package.json:
```diff
@@ -37,9 +37,8 @@
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
peer/dev dependencies to update with tss-react

sbegaudeau replied:
Done

sbegaudeau commented on packages/gantt/frontend/sirius-components-gantt/package.json:
```diff
@@ -31,10 +31,10 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
peer/dev dependencies to update with tss-react

sbegaudeau replied:
Done

sbegaudeau commented on packages/portals/frontend/sirius-components-portals/package.json:
```diff
@@ -32,9 +32,8 @@
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
peer/dev dependencies to update with tss-react

sbegaudeau replied:
Done

sbegaudeau commented on packages/selection/frontend/sirius-components-selection/package.json:
```diff
@@ -33,8 +33,8 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
Same as other package.json

sbegaudeau replied:
Done

sbegaudeau commented on packages/sirius-web/frontend/sirius-web-application/package.json:
```diff
@@ -53,17 +52,17 @@
   "devDependencies": {
     "@apollo/client": "3.10.4",
     "@eclipse-sirius/sirius-components-core": "*",
+    "@eclipse-sirius/sirius-components-deck": "*",
```
tss-react is missing in the dev dependencies

sbegaudeau replied:
Done

sbegaudeau commented on packages/sirius-web/frontend/sirius-web-papaya/package.json:
```diff
@@ -41,9 +41,8 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
+    "@mui/material": "5.15.19",
```
tss-react missing in peer dependencies and mui and tss-react should also be in dev dependencies

sbegaudeau replied:
Done

sbegaudeau commented on packages/sirius-web/frontend/sirius-web/package.json:
```diff
@@ -27,10 +27,9 @@
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
-    "@ObeoNetwork/gantt-task-react": "0.4.19",
+    "@mui/icons-material": "5.15.19",
```
tss-react should be there too

sbegaudeau replied:
Done

sbegaudeau commented on packages/trees/frontend/sirius-components-trees/package.json:
```diff
@@ -34,8 +34,8 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
tss-react

sbegaudeau replied:
Done

sbegaudeau commented on packages/validation/frontend/sirius-components-validation/package.json:
```diff
@@ -32,8 +32,8 @@
-    "@material-ui/core": "4.12.4",
-    "@material-ui/icons": "4.11.3",
+    "@mui/material": "5.15.19",
+    "@mui/icons-material": "5.15.19",
```
tss-react in peer/dev dependencies

sbegaudeau replied:
Done

sbegaudeau commented on vscode-extension/package.json:
```diff
@@ -135,10 +135,9 @@
-    "@material-ui/icons": "4.11.3",
-    "@material-ui/lab": "4.0.0-alpha.61",
-    "@ObeoNetwork/gantt-task-react": "0.4.19",
+    "@mui/icons-material": "5.15.19",
```
tss-react is missing here

sbegaudeau replied:
Done

sbegaudeau commented on packages/core/frontend/sirius-components-core/package.json:
```diff
@@ -33,18 +33,20 @@
     "@xstate/react": "1.6.3",
     "graphql": "16.8.1",
     "react": "17.0.2",
+    "react-dom": "17.0.2",
```
Is react-dom really necessary here? Unless I missed something it should be removed

sbegaudeau replied:
After analysis, it's the other way around, `react-dom` is a peer dependency of `@mui/material` so it should be everywhere (which is currently not the case).

sbegaudeau replied:
Done

sbegaudeau reviewed (changes requested):
I'll take over this PR to fix the last issue that I have and ensure that it gets merged as soon as possible

sbegaudeau commented:
Thanks Denis for the contribution, it took quite some time to integrate it in Sirius Web but it's now finally merged \o/

AresEkb commented:
Thanks a lot for merging!
